papers/bradley_dice/55_bradley_dice.rst

```diff
@@ -0,0 +1,553 @@
+
+A third concept while not new warrants attention in this section, bundling.
+Bundling is a third way to combine pieces of signac-flow's workflow.
+Whereas aggregates are concerned with jobs and groups operations, bundling is concerned with combining what are effectively units of execution into a single submission script.
+The simplest case of a bundle is a submission script with the same operation being executed for multiple jobs either in parallel or serial.
+Bundling is what allows the script to contain multiple jobs executing the same operation.
+By storing information on individual bundles, signac-flow prevents accidental resubmission just as in the unbundled case.
+While the example above does not use either groups or aggregation, bundles works seamlessly with both.
+
+.. figure:: signac-flow_aggregation-groups-bundling.pdf
+    :align: center
+    :scale: 90 %
+    :figclass: w
+
+    Aggregation, groups, and bundling allow users to build complex workflows.
+    The features are orthogonal, and can be used in any combination.
+    Aggregation enables one operation or group to act on multiple jobs.
+    Groups allow users to combine multiple operations into one, with dependencies among operations resolved at run time.
+    Bundling helps users efficiently leverage HPC schedulers by submitting multiple commands in the same script, to be executed in serial or parallel.
+
+Synced Collections: Backend-agnostic, persistent, mutable data structures
+-------------------------------------------------------------------------
+
+Motivation
+~~~~~~~~~~
+
+All of signac's principal functions are designed around efficiently indexing a collection of directories.
+By organizing job directories by the hash of their state point, signac can perform many operations in constant time.
+To present a Pythonic API, state points are exposed via a dictionary-like interface, making it very easy to modify a state point and have that change transparently reflected in both the JSON file and the name of the corresponding directory.
+
+The need to parse these JSON files for indexing and the complexity of modifying them represent the most significant barriers to scaling signac.
+Even in the absence of file modification, reading a large number of files simply to produce a database index becomes prohibitively expensive for large data spaces.
+Although various optimizations have incrementally improved signac's scalability, an alternative means of storing the state point and associated metadata that circumvents the heavy I/O costs of our current approach has the potential to make a much larger impact.
+However, replacing individual JSON files as the primary data source for signac without breaking signac's API would require a generic means for providing the same interface to the underlying index and metadata files irrespective of the underlying storage mechanism.
```

Comment:
We need a replacement for files since we are talking about abstracting out the actual storage mechanism.  I tried "data", but given that metadata is the prior word, it seemed off.

Reply:
This can just say "metadata." It is a noun of its own, not just an adjectival noun / noun modifier.
```suggestion
However, replacing individual JSON files as the primary data source for signac without breaking signac's API would require a generic means for providing the same interface to the underlying index and metadata irrespective of the underlying storage mechanism.
```